enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,609 @@
+
+#### OpenShift API(s)
+
+> Note: The following API(s) will live in the openshift/api repository.
+
+- Introduces a top-level PlatformOperator custom-scoped v1alpha1 CRD:
+
+```go
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:resource:scope=Cluster
+
+// PlatformOperator is the Schema for the platformoperators API.
+type PlatformOperator struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    Spec   PlatformOperatorSpec   `json:"spec,omitempty"`
+    Status PlatformOperatorStatus `json:"status,omitempty"`
+}
+
+// PlatformOperatorSpec defines the desired state of PlatformOperator.
+type PlatformOperatorSpec struct {
+    // PackageName contains the desired OLM-based Operator package name
+    // that is defined in an existing CatalogSource resource in the cluster.
+    //
+    // This configured package will be managed with the cluster's lifecycle.
+    // +kubebuilder:validation:Required
+    PackageName string `json:"packageName"`
+}
+
+// PlatformOperatorStatus defines the observed state of PlatformOperator.
+type PlatformOperatorStatus struct {
+    // Conditions represent the latest available observations of a platform operator's current state.
+    // +optional
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+```
+
+#### Upstream OLM 1.x API(s)
+
+> Note: The following API(s) will live in the openshift/operator-framework-rukpak repository.
+
+- [Introduces the upstream Bundle and BundleDeployment OLM 1.x v1alpha1 API(s)](https://github.com/operator-framework/rukpak/tree/main/api/v1alpha1).
+- [Introduces a validating webhook for the Bundle API](https://github.com/operator-framework/rukpak/blob/main/api/v1alpha1/bundle_webhook.go).
+
+### Risks and Mitigations
+
+#### Risks
+
+- Early phases will introduce multiple v1alpha1 CRDs downstream.
+- Early phases will introduce OLM v1 components that are in alpha stages.
+- No concrete migration plan from legacy to OLM v1.
+- Coordination around external team dependencies.
+- The BundleDeployment API has a large net of permissions.
+- The upstream rukpak project has a direct dependency on cert-manager.
+
+#### Mitigations
+
+The v1alpha1 CRDs won't be deployed by default, and those APIs will be hidden behind the "TechPreviewNoUpgrade" feature set that admins need to explicitly enable before cluster creation. Any new components introduced during these early phases will also specify the `release.openshift.io/feature-gate` annotation in their CVO manifests, and set that key's value to "TechPreviewNoUpgrade".
+This annotation will ensure that the CVO won't deploy any tech preview components.
+
+Instructing admins to configure upfront RBAC to limit the BundleDeployment API to the minimum set of users may help mitigate the plain provisioner's wide net of permissions. The process for constructing this upfront RBAC may require explicit documentation.
```

Comment:
By default, if these are cluster scoped, no one should have access to them apart from cluster admins with wildcard permissions right?

Reply:
That sounds about right. Maybe upfront RBAC isn't necessary here, but I think this behavior is still worth documenting? WDYT.

Reply:
Yeah I think a quick note to alleviate any security concerns is worth adding

Reply:
Updated in https://github.com/openshift/enhancements/pull/1170/commits/cf1de5a212e3c965a73e6af5531a85aa82406a8b.